components/p3a/brave_p2a_protocols.cc

```diff
@@ -0,0 +1,48 @@
+/* Copyright 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/p3a/brave_p2a_protocols.h"
+
+#include <math.h>
+
+#include <numeric>
+#include <vector>
+
+#include "base/rand_util.h"
+
+namespace {
+const double kEpsilon = 2.1;
+}  // namespace
+
+namespace brave {
+
+DirectEncodingProtocol::DirectEncodingProtocol() = default;
+
+DirectEncodingProtocol::~DirectEncodingProtocol() = default;
+
+uint64_t DirectEncodingProtocol::Perturb(
+    uint16_t bucket_count, uint64_t value) {
+  uint64_t perturbed_value = value;
```

Comment:
We probably should have an assert here that `bucket_count > 1`.

Reply:
fixed